### PR TITLE
Change return type of FieldName()

### PR DESCRIPTION
**Pull request by justing-bq**
Resolves https://github.com/amzn/ion-hash-go/issues/46

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hash_reader.go [hunk 1]
```diff
@@ -230,7 +228,13 @@ func (hashReader *hashReader) traverse() error {
 // The following implements hashValue interface.
 
 func (hashReader *hashReader) getFieldName() string {
-	return hashReader.FieldName()
+	fieldName := hashReader.FieldName()
+
+	if fieldName == nil {
```
Comment (Armanbqt) [inline hash_reader.go:233]: We should not return an empty string here.
Nil means fieldname does not exist, whereas "" is a valid fieldname

Comment (justing-bq) [reply hash_reader.go]: Addressed.

File: testing_utilities.go [hunk 2]
```diff
@@ -251,8 +251,8 @@ func decimalStrictEquals(t *testing.T, decimal1, decimal2 *ion.Decimal) {
 func writeFromReaderToWriter(t *testing.T, reader ion.Reader, writer ion.Writer) {
 	for reader.Next() {
 		name := reader.FieldName()
-		if name != "" {
-			require.NoError(t, writer.FieldName(name), "Something went wrong executing writer.FieldName(name)")
+		if name != nil {
```
Comment (Armanbqt) [inline testing_utilities.go:254]: I know it is an existing logic but are you sure it shouldn't be the other way round?
```if name == nil { something went wrong } ```

Comment (therapon) [reply testing_utilities.go:254]: It should be 

* `nil` something went wrong when getting the FieldName() check the reader context for errors 
* non-`nil` you got a fieldName proceed 

https://github.com/amzn/ion-go/pull/93#discussion_r456037974

Comment (justing-bq) [reply testing_utilities.go:254]: The "Something went wrong" part in the require() is referring to something going wrong during writer.FieldName(*name)

Comment (justing-bq) [reply testing_utilities.go:254]: I'll add an assert/require.NotNil() to handle this scenario.

Comment (justing-bq) [reply testing_utilities.go]: Addressed

**Pull request merged**